src/libraries/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestCallback.cs

```diff
@@ -20,6 +23,8 @@ internal static class WinHttpRequestCallback
         public static Interop.WinHttp.WINHTTP_STATUS_CALLBACK StaticCallbackDelegate =
             new Interop.WinHttp.WINHTTP_STATUS_CALLBACK(WinHttpCallback);
 
+        private static ConcurrentDictionary<IPAddress, string> s_cachedCertificates = new();
```

Comment:
This should be specific to each `WinHttpHandler` instance as cert validation callbacks may be different (either an instance collection, or the handler being included as the key).

We should also be clearing old entries from this to avoid a memory leak.